metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MysqlUserMetadataService.java

```diff
@@ -605,6 +635,16 @@ public void saveMetadatas(final String user, final List<? extends HasMetadata> m
                         jdbcTemplate.batchUpdate(SQL.UPDATE_DEFINITION_METADATA, updateDefinitionMetadatas,
                             new int[]{Types.VARCHAR, Types.VARCHAR, Types.VARCHAR});
                     }
+                    if (!insertPartitionDefinitionMetadatas.isEmpty()) {
```

Comment:
Similar as above, we should regroup the partition metadata insert into one function for clarity, or it can be used to update/insert partition definition metadata only.